tools/js-sdk-release-tools/src/mlc/apiVersion/apiVersionTypeExtractor.ts

```diff
@@ -1,81 +1,135 @@
+    });
     return apiVersion;
 };
 
-const getApiVersionTypeFromRestClient: IApiVersionTypeExtractor = (packageRoot: string): ApiVersionType => {
+// workaround for createClient function changes it's way to setup api-version
+export const findApiVersionInRestClient = (clientPath: string): string | undefined => {
+    const version2 = findApiVersionInRestClientV2(clientPath);
+    if (version2) {
+        return version2;
+    }
+    const version1 = findApiVersionInRestClientV1(clientPath);
+    return version1;
+};
+
+const getApiVersionTypeFromRestClient: IApiVersionTypeExtractor = (
+    packageRoot: string
+): ApiVersionType => {
     const clientPath = findRestClientPath(packageRoot);
     const apiVersion = findApiVersionInRestClient(clientPath);
-    if (apiVersion && apiVersion.indexOf("-preview") >= 0) return ApiVersionType.Preview;
-    if (apiVersion && apiVersion.indexOf("-preview") < 0) return ApiVersionType.Stable;
+    if (apiVersion && apiVersion.indexOf("-preview") >= 0)
+        return ApiVersionType.Preview;
+    if (apiVersion && apiVersion.indexOf("-preview") < 0)
+        return ApiVersionType.Stable;
     return ApiVersionType.None;
 };
 
-const findApiVersionsInOperations = (sourceFile: SourceFile | undefined): Array<string> | undefined => {
+const findApiVersionsInOperations = (
+    sourceFile: SourceFile | undefined
+): Array<string> | undefined => {
     const interfaces = sourceFile?.getInterfaces();
-    const interfacesWithApiVersion = interfaces?.filter(itf => itf.getProperty('"api-version"'));
-    const apiVersions = interfacesWithApiVersion?.map(itf => {
-        const property = itf.getMembers()
-            .filter(m => {
-                const defaultValue = m.getChildrenOfKind(SyntaxKind.StringLiteral)[0];
-                return defaultValue && defaultValue.getText() === '"api-version"';
-            })[0];
-        const apiVersion = property.getChildrenOfKind(SyntaxKind.LiteralType)[0].getText();
+    const interfacesWithApiVersion = interfaces?.filter((itf) =>
+        itf.getProperty('"api-version"')
+    );
+    const apiVersions = interfacesWithApiVersion?.map((itf) => {
+        const property = itf.getMembers().filter((m) => {
+            const defaultValue = m.getChildrenOfKind(
+                SyntaxKind.StringLiteral
+            )[0];
+            return defaultValue && defaultValue.getText() === '"api-version"';
+        })[0];
+        const apiVersion = property
+            .getChildrenOfKind(SyntaxKind.LiteralType)[0]
+            .getText();
         return apiVersion;
     });
     return apiVersions;
-}
+};
 
-const getApiVersionTypeFromOperations: IApiVersionTypeExtractor = (packageRoot: string): ApiVersionType => {
-    const paraPath = path.join(packageRoot, 'src/rest/parameters.ts');
+const getApiVersionTypeFromOperations: IApiVersionTypeExtractor = (
+    packageRoot: string
+): ApiVersionType => {
+    const paraPath = path.join(packageRoot, "src/rest/parameters.ts");
```

Comment:
same here, wonder if we should leverage this function for standalone rlc?

Reply:
this function in 'mlc' folder, means it only design for modular client. but we can make the function more general to cover both modular and RLC.